### DISCUSSION TITLE
Vite-SSG cannot build element-plus CSS

### DISCUSSION THREAD

**Discussion by 8bu** [antfu/vite-ssg#232] [General]
## Description

Vite-SSG cannot build element-plus CSS

## Reproduce

- Example source: https://github.com/8bu/vite-ssg-error-element-plus
- Step to reproduce: Run `pnpm build`
- Error:
```bash
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".css" for \\vite-ssg-error-element-plus\node_modules\.pnpm\element-plus@2.1.10_vue@3.2.33\node_modules\element-plus\theme-chalk\el-footer.css
    at new NodeError (node:internal/errors:371:5)
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:87:11)
    at defaultGetFormat (node:internal/modules/esm/get_format:102:38)
    at defaultLoad (node:internal/modules/esm/load:21:14)
    at ESMLoader.load (node:internal/modules/esm/loader:359:26)
    at ESMLoader.moduleProvider (node:internal/modules/esm/loader:280:58)
    at new ModuleJob (node:internal/modules/esm/module_job:66:26)
    at ESMLoader.#createModuleJob (node:internal/modules/esm/loader:297:17)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:261:34)
    at async ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:81:21) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
}
 ELIFECYCLE  Command failed with exit code 1.

```

**Comment by sadeghbarati** [comment]
your repository is private ? 404

**Comment by 8bu** [comment]
sorry my mistake, changed it to public

**Comment by ztxone** [comment]
Same with Swiper:

[vite-ssg] An internal error occurred.
[vite-ssg] Please report an issue, if none already exists: https://github.com/antfu/vite-ssg/issues
node:internal/errors:464
    ErrorCaptureStackTrace(err);
    ^

TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".css" for /my-path/node_modules/.pnpm/swiper@8.1.3/node_modules/swiper/swiper.min.css
    at new NodeError (node:internal/errors:371:5)
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:87:11)
    at defaultGetFormat (node:internal/modules/esm/get_format:102:38)
    at defaultLoad (node:internal/modules/esm/load:21:14)
    at ESMLoader.load (node:internal/modules/esm/loader:359:26)
    at ESMLoader.moduleProvider (node:internal/modules/esm/loader:280:58)
    at new ModuleJob (node:internal/modules/esm/module_job:66:26)
    at ESMLoader.#createModuleJob (node:internal/modules/esm/loader:297:17)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:261:34)
    at async ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:81:21) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
}

**Comment by userquin** [comment]
@ztxone check this https://github.com/antfu/vite-ssg/issues/191#issuecomment-1025166636

**Comment by userquin** [comment]
@8bu I have it working removing `ElementPlusResolver` from unplugin-vue-components plugin options (requires importing element-plus components manually), but then we get this error:
```shell
[vite-ssg] An internal error occurred.
[vite-ssg] Please report an issue, if none already exists: https://github.com/antfu/vite-ssg/issues
file:///F:/work/projects/quini/GitHub/issue-repro/vite-ssg-error-element-plus/node_modules/.pnpm/element-plus@2.1.10_vue@3.2.33/node_modules/element-plus/es/components/popper/src/content.mjs:1
import { placements } from '@popperjs/core';
         ^^^^^^^^^^
SyntaxError: Named export 'placements' not found. The requested module '@popperjs/core' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@popperjs/core';
const { placements } = pkg;
```

**Comment by 8bu** [comment]
> @8bu I have it working removing `ElementPlusResolver` from unplugin-vue-components plugin options (requires importing element-plus components manually), but then we get this error:
> 
> ```shell
> [vite-ssg] An internal error occurred.
> [vite-ssg] Please report an issue, if none already exists: https://github.com/antfu/vite-ssg/issues
> file:///F:/work/projects/quini/GitHub/issue-repro/vite-ssg-error-element-plus/node_modules/.pnpm/element-plus@2.1.10_vue@3.2.33/node_modules/element-plus/es/components/popper/src/content.mjs:1
> import { placements } from '@popperjs/core';
>          ^^^^^^^^^^
> SyntaxError: Named export 'placements' not found. The requested module '@popperjs/core' is a CommonJS module, which may not support all module.exports as named exports.
> CommonJS modules can always be imported via the default export, for example using:
> 
> import pkg from '@popperjs/core';
> const { placements } = pkg;
> ```

Have u try to change vite/ssg build mode to cjs?
Btw I wanna use the element-plus resolver then so need to wait for a better implementation or vite-ssg fix this issue

**Comment by userquin** [comment]
@8bu patching `node_modules/element-plus/es/components/popper/src/content.mjs` seems to work, removing also `ElementPlusResolver` from `unplugin-vue-components` plugin option on your repo:
```js
//import { placements } from '@popperjs/core';
import * as pkg from '@popperjs/core';
const { placements } = pkg;
```
![imagen](https://user-images.githubusercontent.com/6311119/164500010-f475a658-01ac-4ed8-8650-01171aa6087b.png)

**Comment by userquin** [comment]
> Have u try to change vite/ssg build mode to cjs?

Yes, the same error

> Btw I wanna use the element-plus resolver then so need to wait for a better implementation or vite-ssg fix this issue

We cannot fix it here

**Reply by 8bu** [reply]
I guess I will not use SSG then since the resolver make things easier

**Comment by userquin** [comment]
@8bu running the `https-preview` script:

![imagen](https://user-images.githubusercontent.com/6311119/164501373-95e73ad5-4e85-4c57-b0ef-47f7b4a09ac0.png)

**Reply by 8bu** [reply]
it looks broken (the aside is placed on top). Probably no css was imported at all

**Reply by userquin** [reply]
adding the resolver and running the dev script the layout is the same

**Reply by userquin** [reply]
you're right: I forget to remove the static imports

**Comment by matiboy** [comment]
`element-plus@2.1.11` fixes the popperjs import issue (https://github.com/element-plus/element-plus/pull/7321)
Unplugin resolver not working atm but manually importing the components allows to build successfully. 
Need to import the css as well, per the documentation `import 'element-plus/dist/index.css'` in `main.ts`

**Comment by akil-rails** [comment]
Unplugin resolver still not working, have to import the components & css manually .

**Comment by brc-dd** [comment]
You probably need to pass `{ ssr: true }` to `ElementPlusResolver`. Here is the output on your repo (with updated deps):

![image](https://user-images.githubusercontent.com/40380293/184545173-6b81d1b6-8e59-414a-88d0-8239f035b326.png)

But quite large chunks are being generated:

![image](https://user-images.githubusercontent.com/40380293/184545288-c75eb9fc-23fa-4dcc-b27f-4ee31b19683e.png)

Is tree-shaking not working properly? I can count over 150 components in that file.

**Comment by socketbear** [comment]
https://github.com/antfu/vite-ssg/issues/171#issuecomment-1212862501

solution, here!
